apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx

```diff
@@ -76,36 +76,52 @@ const useHandleOnChange = (
   }, [value]);
 };
 
-const useScrub = (options: {
+const useScrub = ({
+  value,
+  onChange,
+  onChangeComplete,
+}: {
   value: StyleValue;
   onChange: (value: StyleValue) => void;
   onChangeComplete: (value: StyleValue) => void;
 }) => {
   const inputRef = useRef<HTMLInputElement | null>(null);
-  const optionsRef = useRef(options);
-  optionsRef.current = options;
+  const onChangeRef = useRef(onChange);
+  onChangeRef.current = onChange;
+  const onChangeCompleteRef = useRef(onChangeComplete);
+  onChangeCompleteRef.current = onChangeComplete;
+  const valueRef = useRef(value);
+  valueRef.current = value;
+
+  const type = valueRef.current.type;
+  const unit = type === "unit" ? valueRef.current.unit : undefined;
```

Comment:
This code should be identical to:

```
const type = value.type;
const unit = type === "unit" ? value.unit : undefined;
```

Because the line above does `valueRef.current = value`.

Otherwise this is more or less what I meant, yeah :)